src/include/server/mir/default_server_configuration.h

```diff
@@ -459,7 +459,16 @@ class DefaultServerConfiguration : public virtual ServerConfiguration
         seat_observer_multiplexer;
     CachedPtr<ObserverMultiplexer<frontend::SessionMediatorObserver>>
         session_mediator_observer_multiplexer;
-    std::shared_ptr<report::Reports> const reports;
+
+    // We should to defer initializing reports until any subclasses have been
+    // fully constructed as that allows them to configure the reporting and/or
+    // logging options.
+    // So we initializing reports in the_options() which will certainly be
+    // called before starting the server. The only tricky bit is that the_options()
+    // is called when initializing reports. Hence we need a flag to avoid endless
+    // recursion.
+    bool mutable the_options_has_been_called{false};
```

Comment:
Seems like excessive complexity. That being said, I don't have a proposal for how to do it better.